Can't `pub use` a test function marked with `#[serial]` in a private module

I'm trying to export a specific `#[test]` function so it's available outside of the enclosing `tests` module (which is private) using `pub use`, so a different test in a different file can run it. However, when I add the `#[serial]` attribute to the function, the `pub use` stops working and I get this compiler error:
```
error[E0603]: function `test` is private
 --> src/main.rs:1:16
  |
1 | pub use tests::test;
  |                ^^^^ private function
  |
note: the function `test` is defined here
 --> src/main.rs:6:5
  |
6 |     #[serial]
  |     ^^^^^^^^^
  = note: this error originates in the attribute macro `serial` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Here's `main.rs`:
```rust
pub use tests::test;

mod tests {
    use serial_test::serial;

    #[serial]
    pub fn test() {}
}
```

`rustc --version --verbose`

```
rustc 1.67.0-nightly (42325c525 2022-11-11)
binary: rustc
commit-hash: 42325c525b9d3885847a3f803abe53c562d289da
commit-date: 2022-11-11
host: x86_64-apple-darwin
release: 1.67.0-nightly
LLVM version: 15.0.4
```

--- a/serial_test_derive/src/lib.rs
+++ b/serial_test_derive/src/lib.rs
@@ -504,6 +504,23 @@ mod tests {
     }
 
     #[test]
+    fn test_serial_with_pub() {
+        let attrs = proc_macro2::TokenStream::new();
+        let input = quote! {
+            #[test]
+            pub fn foo() {}
+        };
+        let stream = local_serial_core(attrs.into(), input);
+        let compare = quote! {
+            #[test]
+            pub fn foo () {
+                serial_test::local_serial_core("",  :: std :: option :: Option :: None, || {} );
+            }
+        };
+        assert_eq!(format!("{}", compare), format!("{}", stream));
+    }
+
+    #[test]
     fn test_serial_with_timeout() {
         let attrs = vec![
             TokenTree::Ident(format_ident!("timeout_ms")),

--- a/serial_test_derive/src/lib.rs
+++ b/serial_test_derive/src/lib.rs
@@ -377,6 +377,7 @@ where
     if asyncness.is_some() && cfg!(not(feature = "async")) {
         panic!("async testing attempted with async feature disabled in serial_test!");
     }
+    let vis = ast.vis;
     let name = ast.sig.ident;
     let return_type = match ast.sig.output {
         syn::ReturnType::Default => None,
@@ -414,7 +415,7 @@ where
                 quote! {
                     #(#attrs)
                     *
-                    async fn #name () -> #ret {
+                    #vis async fn #name () -> #ret {
                         serial_test::#fnname(#(#args ),*, || async #block ).await;
                     }
                 }
@@ -424,7 +425,7 @@ where
                 quote! {
                     #(#attrs)
                     *
-                    fn #name () -> #ret {
+                    #vis fn #name () -> #ret {
                         serial_test::#fnname(#(#args ),*, || #block )
                     }
                 }
@@ -437,7 +438,7 @@ where
                 quote! {
                     #(#attrs)
                     *
-                    async fn #name () {
+                    #vis async fn #name () {
                         serial_test::#fnname(#(#args ),*, || async #block ).await;
                     }
                 }
@@ -447,7 +448,7 @@ where
                 quote! {
                     #(#attrs)
                     *
-                    fn #name () {
+                    #vis fn #name () {
                         serial_test::#fnname(#(#args ),*, || #block );
                     }
                 }

